crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -114,7 +124,23 @@ declare_lint! {
 
 declare_lint! {
     /// ## What it does
-    /// Checks whether an argument is used as both a value and a type form in a call
+    /// Checks whether an argument is used as both a value and a type form in a call.
+    ///
+    /// ## Why is this bad?
+    /// Such calls have confusing semantics and often indicate a logic error.
```

Comment:
Can we add (or link to) a definition of "type form" somewhere here? I think most users will be unfamiliar with the term

Reply:
Yeah. You could maybe switch to using "type expression" instead of "type form"? That is well defined -- you could link to https://typing.python.org/en/latest/spec/annotations.html#type-and-annotation-expressions

Reply:
Okay, let's leave this for now